dbt/adapters/snowflake/relation_configs/dynamic_table.py

```diff
@@ -21,6 +35,147 @@ class SnowflakeDynamicTableConfig(RelationConfigBase):
     """
 
     name: str
+    schema_name: str
+    database_name: str
     query: str
     target_lag: SnowflakeDynamicTableTargetLagConfig
     warehouse: str
+
+    @property
+    def path(self) -> str:
+        return ".".join(
+            part for part in [self.database_name, self.schema_name, self.name] if part is not None
+        )
+
+    @property
+    def schema_path(self) -> str:
+        return ".".join(
+            part for part in [self.database_name, self.schema_name] if part is not None
+        )
+
+    @property
+    def validation_rules(self) -> Set[RelationConfigValidationRule]:
+        return super().validation_rules
+
+    @classmethod
+    def from_dict(cls, config_dict: dict) -> "SnowflakeDynamicTableConfig":
+        kwargs_dict = {
+            "query": config_dict.get("query"),
+            "warehouse": config_dict.get("warehouse"),
+        }
+
+        if name := config_dict.get("name"):
+            kwargs_dict.update({"name": cls._render_part(ComponentName.Identifier, name)})
+        if schema_name := config_dict.get("schema_name"):
+            kwargs_dict.update(
+                {"schema_name": cls._render_part(ComponentName.Schema, schema_name)}
+            )
+        if database_name := config_dict.get("database_name"):
+            kwargs_dict.update(
+                {"database_name": cls._render_part(ComponentName.Database, database_name)}
+            )
+
+        if target_lag := config_dict.get("target_lag"):
+            kwargs_dict.update(
+                {"target_lag": SnowflakeDynamicTableTargetLagConfig.from_dict(target_lag)}
+            )
+
+        dynamic_table: "SnowflakeDynamicTableConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return dynamic_table
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        config_dict = {
+            "name": model_node.identifier,
+            "schema_name": model_node.schema,
+            "database_name": model_node.database,
+            "query": model_node.compiled_code,
+            "target_lag": SnowflakeDynamicTableTargetLagConfig.parse_model_node(model_node),
+            "warehouse": model_node.config.extra.get("warehouse"),
+        }
+        return config_dict
+
+    @classmethod
+    def parse_relation_results(cls, relation_results: RelationResults) -> dict:
```

Comment:
Where does this method get used? For output or something else?

Reply:
It get's used in `super().from_relation_results()`, which merely chains together this and `from_dict()`. This may not make as much sense here, but I'm going for a consistent look/feel across adapters and `RelationConfigBase` classes. When there are attributes which are themselves `RelationConfigBase` classes, e.g. `target_lag` below, then we need this method as a stopping point to produce a `dict`. Hence it gets called if this instance were an attribute on a bigger object.

tl;dr; Consistency? And to fit into the base class framework